app/scripts/models/reliers/oauth.js

```diff
@@ -140,10 +146,10 @@ define(function (require, exports, module) {
       self._importRequiredSearchParam('client_id', 'clientId');
       self._importRequiredSearchParam('scope', 'scope');
       self.importSearchParam('state');
-      self.importSearchParam('redirect_uri', 'redirectUri');
+      self._importValidateOptionalSearchParam('redirect_uri', 'redirectUri', Validate.isUriValid, OAuthErrors);
```

Comment:
> It seems like I need to make the validation functionality a little bit more flexible so that it can be used for more than just search params. Something like Joi but maybe more compact. Or it might be possible to extend isDataValid to support testing for values?

This is almost the exact conversation @philbooth and I had, which is where `isDataValid` came from. We even looked at using joi in the browser (it seems possible, though we didn't try it). 

I want to move in that direction, though not in this PR. Let's keep this in mind and figure out a good declarative way of importing, renaming and validating.